File: app/src/main/java/com/tjb/dwf/PinchGestureReceiver.java
package com.tjb.dwf;

import android.util.Log;

public class PinchGestureReceiver {
    private MainActivity mainActivity;

    public void installMainActivity(MainActivity mainActivity) {
        this.mainActivity = mainActivity;
    }

    public void detachMainActivity() {
        this.mainActivity = null;
    }

    public void showOptions() {
        if (mainActivity != null) {
            mainActivity.showOptions();
        } else {
            Log.e("PinchGestureReceiver", "MainActivity is null");
        }
    }

    public void showPicture() {
        if (mainActivity != null) {
            mainActivity.showPicture();
        } else {
            Log.e("PinchGestureReceiver", "MainActivity is null");
        }
    }
}

File: app/src/main/java/com/tjb/dwf/main/PinchGestureReceiver.kt
package com.tjb.dwf.main

import android.util.Log
import com.tjb.dwf.di.ActivityScope
import javax.inject.Inject

@ActivityScope
class PinchGestureReceiver @Inject constructor() {
    private var mainActivity: MainActivity? = null

    fun installMainActivity(mainActivity: MainActivity) {
        this.mainActivity = mainActivity
    }

    fun uninstallMainActivity() {
        this.mainActivity = null
    }

    fun showOptions() {
        // note, all this nullable crap should be not an issue once this is scoped to main activities lifecycle
        if (mainActivity != null) {
            mainActivity?.showOptions()
        } else {
            Log.e("PinchGestureReceiver", "MainActivity is null")
        }
    }

    fun showPicture() {
        // note, all this nullable crap should be not an issue once this is scoped to main activities lifecycle
        if (mainActivity != null) {
            mainActivity?.showPicture()
        } else {
            Log.e("PinchGestureReceiver", "MainActivity is null")
        }
    }
}
